lib/repl.js

```diff
@@ -434,22 +435,31 @@ function REPLServer(prompt,
       }
     }
 
-    if (errStack === '') {
-      errStack = `Thrown: ${util.inspect(e)}\n`;
-    } else {
-      const ln = errStack.endsWith('\n') ? '' : '\n';
-      errStack = `Thrown:\n${errStack}${ln}`;
-    }
-
     if (!self.underscoreErrAssigned) {
       self.lastError = e;
     }
 
     const top = replMap.get(self);
-    top.outputStream.write(errStack);
-    top.clearBufferedCommand();
-    top.lines.level = [];
-    top.displayPrompt();
+    if ((process._forceRepl || isatty(0)) &&
+        process.listenerCount('uncaughtException') !== 0) {
+      process.nextTick(() => {
```

Comment:
why is this deferred by a nextTick?

Reply:
It would otherwise end up in a weird domain hick up: the domain would catch the here emitted exception.

Reply:
See also https://github.com/nodejs/node/pull/20803#discussion_r238120518